src/Forms/SearchableDropdownTrait.php

```diff
@@ -0,0 +1,567 @@
+        $this->labelField = $labelField;
+        return $this;
+    }
+
+    public function getAttributes(): array
+    {
+        $name = $this->getName();
+        if ($this->isMultiple) {
+            $name .= '[]';
+        }
+        return array_merge(
+            parent::getAttributes(),
+            [
+                'name' => $name,
+                'data-schema' => json_encode($this->getSchemaData()),
+            ]
+        );
+    }
+
+    /**
+     * Get a list of selected ID's
+     */
+    public function getValueArray(): array
+    {
+        $value = $this->Value();
+        if (empty($value)) {
+            return [];
+        }
+        if (is_array($value)) {
+            $arr = $value;
+            // Normalise FormBuilder values to be like Page EditForm values
+            //
+            // FormBuilder $values for non-multi field will be
+            // [
+            //   'label' => 'MyTitle15', 'value' => '10'
+            // ]
+            if (array_key_exists('value', $arr)) {
+                $val = (int) $arr['value'];
+                return $val ? [$val] : [];
+            }
+            // FormBuilder $values for multi will be
+            // [
+            //   0 => ['label' => '10', 'value' => 'MyTitle10', 'selected' => false],
+            //   1 => ['label' => '15', 'value' => 'MyTitle15', 'selected' => false]
+            // ];
+            $firstKey = array_key_first($arr);
+            if (is_array($arr[$firstKey]) && array_key_exists('value', $arr[$firstKey])) {
+                $newArr = [];
+                foreach ($arr as $innerArr) {
+                    $val = (int) $innerArr['value'];
+                    if ($val) {
+                        $newArr[] = $val;
+                    }
+                }
+                return $newArr;
+            }
+            // Page EditForm $values for non-multi field will be
+            // [
+            //   0 => '10',
+            // ];
+            // Page EditForm $values for multi will be
+            // [
+            //   0 => '10',
+            //   1 => '15'
+            // ];
+            return array_map('intval', $arr);
+        }
+        if ((is_string($value) || is_int($value)) && ctype_digit((string) $value) && $value != 0) {
+            return [(int) $value];
+        }
+        if ($value instanceof SS_List) {
+            return array_filter($value->column('ID'));
+        }
+        if ($value instanceof DataObject && $value->exists()) {
+            return [$value->ID];
+        }
+        // Don't know what value is, handle gracefully. We should not raise an exception here because
+        // of there is a bad data for whatever a content editor will not be able to resolve and it will
+        // render part of the CMS unusable
+        // /** @var LoggerInterface $logger */
+        $logger = Injector::inst()->get(LoggerInterface::class);
+        $logger->warning('Could not determine value in ' . $this::class . '::getValueArray()');
```

Comment:
```suggestion
        $logger->warning('Could not determine value in ' . static::class . '::getValueArray()');
```
Or just use `__CLASS__`

Reply:
Updated